src/key.rs

```diff
@@ -212,9 +212,9 @@ impl SecretKey {
         SecretKey(sk)
     }
 
-    /// Serializes the secret key as byte value.
+    /// Gets the secret key as a byte value.
     #[inline]
-    pub fn serialize_secret(&self) -> [u8; constants::SECRET_KEY_SIZE] {
+    pub fn into_secret(&self) -> [u8; constants::SECRET_KEY_SIZE] {
```

Comment:
In 2ccd54bb579c07a71c3610da8955f3b9fb3e8811:

I don't know about this. `into_secret` I would expect to produce a `SecretKey` or maybe a `SharedSecret`, not a byte array. A byte array I'd expect to get from something whose name was serialization-related.

Reply:
Something bothered me about this and now I see what. `secret.into_secret()` is weird. Even worse it takes `&self`, which shouldn't use `into_` prefix. I think the name should stay or `_secret` should be removed. Or perhaps call it `to_bytes()` to remove the impression that there's costly serialization.

Reply:
Nice. I'd be happy with `to_bytes`.

Reply:
The system works, `to_bytes` it is.

Reply:
We need to write our own, non-confusing version. :)